support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -233,3 +250,81 @@ object CreateZuoraSubscription {
     }
 
 }
+
+object DigitalSubscriptionGiftRedemption {
+
+  def ifDigitalSubscriptionGiftRedemption(product: ProductType, paymentMethod: Either[PaymentMethod, RedemptionData]) = {
+    product match {
+      case d: DigitalPack if paymentMethod.isRight && d.readerType == Gift => paymentMethod.right.toOption
+      case _ => None
+    }
+  }
+
+  def redeemGift(
+    redemptionData: RedemptionData,
+    gifteeUserId: String,
+    requestInfo: RequestInfo,
+    state: CreateZuoraSubscriptionState,
+    zuoraService: ZuoraService,
+    catalogService: CatalogService
+  ): Future[HandlerResult[SendThankYouEmailState]] = {
+    for {
+      giftSubscriptionFields <- zuoraService.getSubscriptionFromRedemptionCode(redemptionData.redemptionCode)
+      validatedSubscription <- Future.fromTry(validateRedemptionData(giftSubscriptionFields))
+      fullGiftSubscription <- zuoraService.getSubscriptionById(validatedSubscription.id)
+      newTermLength <- Future.fromTry(calculateNewTermLength(fullGiftSubscription, catalogService))
+      updateDataResponse <- zuoraService.updateSubscriptionRedemptionData(validatedSubscription.id, gifteeUserId, newTermLength)
```

Comment:
is there a better name for `updateDataResponse` or is it literally just in case something strange is returned we want to save it?

Reply:
I think it's just success false/true and a possible error message, so happy to resolve this one